### ISSUE TITLE
Mouse interactions do not appear to be working properly on windows

### ISSUE THREAD

**Issue by zacps** [alacritty/alacritty#1663]
[Original issue](https://github.com/zacps/alacritty/issues/19) by @dparnell

Which operating system does the issue occur on?

Windows 10 client connecting to a Linux 16.04 machine via ssh

Steps to reproduce:

1. Make sure that `dialog` is installed
2. Execute `dialog --yesno Testing 10 30`
3. Attempt to use the mouse to click on either the Yes or No button. Nothing happens

Expected bahavior:

Clicking on either of the buttons should cause the `dialog` process to exit

**Comment by zacps** [comment]
From @parthsdoshi

Here's a test case that works using cmd.exe but not with alacritty.exe:

1. Run `bash`
2. Run `tmux`
3. Make sure tmux mouse mode is on: `<Leader> :set -g mouse on`
4. Run `dialog --yesno Testing 10 30`
5. Verify clicking works using cmd.exe
6. Detach tmux: `<Leader> d`
7. open alacritty.exe
8. Run `bash`
9. Run `tmux a`
10. Run `dialog --yesno Testing 10 30`
11. Verify the same does not work in alacritty.

There's a few simpler test cases too.
Open bash through cmd.exe and test vim scrolling. Do the same in alacritty.exe and see that vim won't receive any mouse input.

**Comment by kumarharsh** [comment]
I've found that scrolling in Alacritty scrolls the whole alacritty buffer. So if I'm running tmux, the tmux panes themselves scroll off-view when I scroll in _any_ pane.

I then opened ConEMU and attached to the same tmux session, and I am able to scroll in the vim editor pane. It still doesn't scroll in panes which have other programs (say webpack, etc) running in them. The whole screen doesn't scroll like alacritty, but the tmux pane hovered over receives the mouse events and starts outputting `^[[A^[[A^[[A^[[A^[[A^[[A^[[A^[[A` characters instead of scrolling.

**Comment by mqudsi** [comment]
Yes, that's to be expected because the scroll event is not being sent to the application, only to the terminal's pager.

**Comment by mqudsi** [comment]
Perhaps @zacps  or a team member can update the original issue with a link to this, which I suppose can be considered the upstream issue: https://github.com/Microsoft/console/issues/376

@DHowett-MSFT has explained why this doesn't work (it's a ConPTY thing, nothing to do with alacritty one bit)

**Comment by malaverdiere** [comment]
This is happening to me with `enable_experimental_conpty_backend` set to both `true` and `false`. So maybe this is more than a ConPTY issue.

**Comment by kumarharsh** [comment]
Mouse scrolling does work in `wsl.exe`, so I too think might be an Alacritty thing?

**Comment by mqudsi** [comment]
@kumarharsh that's because `wsl.exe` (and `bash.exe`) is not a conpty process.

**Comment by malaverdiere** [comment]
This is happening to me with `enable_experimental_conpty_backend` set to both `true` and `false`. So maybe this is more than a ConPTY issue.

**Comment by kumarharsh** [comment]
Yeah, I kinda realised that just after posting, so I deleted my comment to prevent confusion (sorry about that)

**Comment by llinfeng** [comment]
I find mouse clicks to be accepted for WSL-tmux (on Windows) via [wsltty](https://github.com/mintty/wsltty). Source: https://github.com/microsoft/terminal/issues/545#issuecomment-506876814

Are there special settings for `alacritty` to enable such mouse features? On [wsltty](https://github.com/mintty/wsltty), I even found mouse-clicks to be working for SSH sessions.

**Comment by chrisduerr** [comment]
> Are there special settings for alacritty to enable such mouse features?

No, the shell is likely just not sending/accepting the right escapes.

**Comment by unphased** [comment]
To confirm, mouse is just not working yet in alacritty on windows? I can only select with it.

**Comment by kchibisov** [comment]
Ehm, what else do you want from it? Are you talking about `mouse mode`?

**Comment by unphased** [comment]
@kchibisov yes that's what I meant, sorry.

**Comment by kchibisov** [comment]
Could you try running this `echo -e "\e[?1000". and after that clicking with your mouse. it should paste random garbage.

P.s. I'm not sure whether windows has `echo -e` thing, but you should get an idea on what to do.

**Comment by DHowett-MSFT** [comment]
If Alacritty is using the windows pseudoconsole infrastructure (not wsltty(!)), you're still subject to the limitations mentioned in https://github.com/microsoft/terminal/issues/545#issuecomment-506876814. The system-level support for translating mouse events isn't there just yet. :smile:

**Comment by kchibisov** [comment]
I've just found out a link to a thread, thx @DHowett-MSFT .

**Comment by tcneuen** [comment]
Latest version of MS terminal infrastructure has been updated [updated](https://github.com/microsoft/terminal/issues/545#) to handle mouse events

**Comment by DHowett-MSFT** [comment]
Just so you're aware: the console host changes here will be released in an upcoming version of Windows (probably not the most immediate one if the build number is <19500), and _the changes we made do not support mouse events for Windows applications_.

For anyone working on a terminal: The pty hosting infrastructure now hooks the terminal's input stream directly to the application's input stream; there's no more input translation in the "pure VT" case.

We still have a lot of work to do to synthesize windows mouse events for windows applications where we _do_ need translation, but this may be enough to unblock other ConPTY consumers.

**Comment by kchibisov** [comment]
@DHowett-MSFT 

So If I understand you correctly, if we have perfectly working mouse mode on all platforms except Windows(due to Windows), after this update things should work right away in ConPty for us?

**Comment by DHowett-MSFT** [comment]
Yes, if the application on the other end is in VT input mode (so is running in WSL, newer versions of Cygwin/MSYS2, or otherwise using the `ENABLE_VIRTUAL_TERMINAL_INPUT` handle mode as a Win32 application.)

**Comment by kchibisov** [comment]
Ok, thank you.

**Comment by damnskippy** [comment]
@DHowett-MSFT 
> the console host changes here will be released in an upcoming version of Windows

Sorry if I'm being dense - the note above seems to suggest the mouse input related enhancement is available through the upcoming version of Windows.  However my interpretation may not be entirely accurate since latest release Terminal Preview advertises mouse input support which I presume runs the the current Windows release?.  I'm not sure.   Or are the mouse related enhancements somehow bundled into the Terminal Preview.

For an alacritty user like me on Windows, it sure sounds like waiting for the next Windows release is the only recourse.  (Sadly IT at my work is slow to roll out windows upgrades, so seeing mouse input on Alacritty for me is so near and yet so far).

Thanks.

**Comment by DHowett-MSFT** [comment]
@damnskippy sorry about the confusion! Terminal ships with a bundled version of conhost so that it can leverage improvements to the pty.

**Comment by damnskippy** [comment]
Thank you, @DHowett-MSFT - that clears it up!

**Comment by kchibisov** [comment]
Since alacritty can't do anything other than wait, closing this issue.

**Comment by PetterS** [comment]
@kchibisov why close it? Isn't it better to track this?

**Comment by Gskartwii** [comment]
I think this issue should be opened, so that it shows up more clearly in the search, preventing duplicates.

**Comment by kchibisov** [comment]
>I think this issue should be opened, so that it shows up more clearly in the search, preventing duplicates.

Just search closed issues as well, we can't do anything about mouse on Windows, since it's not something depending on us. It should just start working at some point, since mouse protocol is implemented for a long time in alacritty.

**Comment by habamax** [comment]
Still doesn't work with windows 10 2004.

![image](https://user-images.githubusercontent.com/234774/84370155-8b851480-abe0-11ea-970e-491220c919f5.png)

**Comment by Neur0toxine** [comment]
@habamax  I replaced `conhost.exe` with `OpenConsole.exe` from Windows Terminal (microsoft/terminal#1817). It works just fine.
Don't forget to restore original permissions (TrustedInstaller as owner, read and execute permission for all other groups & users).

**Comment by habamax** [comment]
> 
> 
> @habamax I replaced `conhost.exe` with `OpenConsole.exe` from Windows Terminal ([microsoft/terminal#1817](https://github.com/microsoft/terminal/issues/1817)). It works just fine.
> Don't forget to restore original permissions (TrustedInstaller as owner, read and execute permission for all other groups & users).

Thx, although I am not sure what is conhost.exe and how can I replace it with openconsole.exe

**Comment by niklaskorz** [comment]
@Neur0toxine that worked, thanks a lot! :) Perfect for bridging the time until October/November when Windows 10 hopefully uses OpenConsole as its default conhost.

**Comment by DHowett** [comment]
> @Neur0toxine that worked, thanks a lot! :) Perfect for bridging the time until October/November when Windows 10 hopefully uses OpenConsole as its default conhost.

conhost is built from the (exact!) same source as OpenConsole; one is just delivered a lot faster since we aren’t stuck behind the multiple-month Windows release cycle ;)

**Comment by bernastoc** [comment]
I replaced conhost.exe as suggested. While 'echo -e "\e[?1000h"' results in garbage characters on mouse inputs, tmux doesn't capture them. Of course I have mouse enabled in tmux and vim (to validate my tmux/vim setup I tried with Terminus and mouse worked as expected).

**Comment by kchibisov** [comment]
Tmux requires `echo -e "\e[?1003h"` IIRC or `echo -e "\e[?1002h"`

**Comment by bernastoc** [comment]
Thanks. It turns out when I launch Alacritty with '-e ssh' the remote tmux/vim sessions don't capture the mouse. Launching Alacritty without options, _then_ initiating an ssh session results in the desired mouse capture I need in tmux. The same issue exists in WT and it's apparently due to https://github.com/PowerShell/Win32-OpenSSH/issues/1310.

**Comment by HHPLow** [comment]
> @habamax I replaced `conhost.exe` with `OpenConsole.exe` from Windows Terminal ([microsoft/terminal#1817](https://github.com/microsoft/terminal/issues/1817)). It works just fine.
> Don't forget to restore original permissions (TrustedInstaller as owner, read and execute permission for all other groups & users).

Thank you very much, by the way, `OpenConsole.exe` can be found from the installation location of WindowTerminal, it should be updated more frequently than Windows XD.
`C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_<VERSION information>\OpenConsole.exe`
For example 
`C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_1.0.1811.0_x64__8wekyb3d8bbwe\OpenConsole.exe`

**Comment by ludwj61** [comment]
@HHPLow Would you mind explaining how to get alacritty to use OpenConsole? Are we talking replacing the conhost in the windows files or something else?

**Comment by bernastoc** [comment]
> @HHPLow Would you mind explaining how to get alacritty to use OpenConsole? Are we talking replacing the conhost in the windows files or something else?

Exactly. Copy openconsole.exe it to C:\Windows\System32 and rename it to conhost.exe (probably should save the original conhost though, just to be safe).

**Comment by ludwj61** [comment]
@bernastoc Thanks that did the trick.

**Comment by quinn** [comment]
> Exactly. Copy openconsole.exe it to C:\Windows\System32 and rename it to conhost.exe (probably should save the original conhost though, just to be safe).

Hi, i want to confirm this is the entire process necessary to get this to work? or are there other steps? Currently, i start alacritty, then run wsl, then run tmux. is the above step all i would need to get the scroll wheel, etc working?

**Comment by HHPLow** [comment]
> > Exactly. Copy openconsole.exe it to C:\Windows\System32 and rename it to conhost.exe (probably should save the original conhost though, just to be safe).
> 
> Hi, i want to confirm this is the entire process necessary to get this to work? or are there other steps? Currently, i start alacritty, then run wsl, then run tmux. is the above step all i would need to get the scroll wheel, etc working?

1. Close all running cmd program, like cmd or running alacritty.
2. Rename your original `conhost.exe` to other names like `conhost.exe_bak` in `C:\Windows\System32\`
3. Go to `Windows Terminal` path `C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_<VERSION information>\OpenConsole.exe`  
**e.g.**:`C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_1.0.1811.0_x64__8wekyb3d8bbwe\OpenConsole.exe`
4. Copy and rename `OpenConsole.exe` to `C:\Windows\System32\conhost.exe`
5. Reopen your alacrity and open wsl then tmux, you can enjoy~
BTW, please make sure you keep `winpty_backend: false` in alacrity config.

And you can check this by using windows task manager
![image](https://user-images.githubusercontent.com/10094017/97065724-21469c80-15e2-11eb-96d0-93045c1f2978.png)

**Comment by fredizzimo** [comment]
The issues were not fixed on the Windows 20H2 build, at least not for me. And worse, I was no longer able to replace the conhost.exe, since it starts with Windows, and there's no way to kill it. 

So I made this pull request, https://github.com/alacritty/alacritty/pull/4501, which allows you to put the conhost.dll, and OpenConsole.exe files of the Windows Terminal into the Alacritty folder, without having to mess any system files.

**Comment by ndavidpsi** [comment]
Any updates?

**Comment by ric2work** [comment]
Hi  

Tq so much for you still remember my case and email to me ! 

Apple  technical  said they  cnt  help , after I 4 times walk in !




Ric Tan 
spaces zone design creation 
+60164808297

**Comment by cki0** [comment]
Is there no progress on this? 

Mouse interactions work perfectly fine in `wsl+tmux+neovim` running on cmd.exe. Works perfectly running on Windows Terminal. And it works running on wsltty. Alacritty is now the odd one out here with mouse events not working.

**Comment by NathanKong214** [comment]
> 
> 
> Is there no progress on this?
> 
> Mouse interactions work perfectly fine in `wsl+tmux+neovim` running on cmd.exe. Works perfectly running on Windows Terminal. And it works running on wsltty. Alacritty is now the odd one out here with mouse events not working.

Got here from Google search. Seems not. I was trying to click in `htop` to sort before noticing it doesn't work. It works fine on Windows Terminal, CMD, PowerShell, and `wsl.exe`, with and without `tmux`

I'm running Windows 10 build 21H1 if that helps.

**Comment by Viqsi** [comment]
I'm not sure I get the rationale behind this decision, particularly since WinPTY support was removed last year. If you're not going to support WinPTY (the old console system on Windows), and you're not going to support ConPTY at the level literally everybody else in the Windows space does (the current console system on Windows), one frankly has to wonder how/why the Windows port exists at all, given its status of being pointlessly crippled.

**Comment by prwang** [comment]
> Is there no progress on this?
> 
> Mouse interactions work perfectly fine in `wsl+tmux+neovim` running on cmd.exe. Works perfectly running on Windows Terminal. And it works running on wsltty. Alacritty is now the odd one out here with mouse events not working.

See discussions above on replacing C:\Windows\system32\conhost.exe with OpenConsole.exe. It works for me.

**Comment by kahnclusions** [comment]
> > Is there no progress on this?
> > 
> > Mouse interactions work perfectly fine in `wsl+tmux+neovim` running on cmd.exe. Works perfectly running on Windows Terminal. And it works running on wsltty. Alacritty is now the odd one out here with mouse events not working.
> 
> See discussions above on replacing C:\Windows\system32\conhost.exe with OpenConsole.exe. It works for me.

I wouldn't consider that a solution, no one should need to mess with files in system32 just to make a terminal emulator work. I've switched to Windows Terminal for now, it works a lot better than Alacrity. Recommend others do the same.

**Comment by fredizzimo** [comment]
A much less hacky solution is to build my branch https://github.com/fredizzimo/alacritty/tree/add_openconsole_support, then build the Windows terminal, following these instructions https://github.com/microsoft/terminal#developer-guidance. 

After that you can copy conpty.dll and openconsole.exe from the windows terminal binaries into the same directory as your custom version of Alacritty. They will then be automatically loaded when you launch Alacritty and the mouse and other things should work as they should. And if you for some reason don't want to load them anymore, you can just delete them, and Alacritty will use the default Windows implementation like before.

I made a pull request a long time ago https://github.com/alacritty/alacritty/pull/4501, but the maintainers of Alacritty are totally unwilling to accept the changes. Somehow they have problems accepting a library provided by Microsoft, which is used in virtually all other Windows terminal emulators, just because Windows includes an older version of it by default.  

BTW, you might want to merge a newer version of Alacritty on top of my branch, since I haven't updated it for a while. And I doubt that I will update it in the future, unless the the project maintainers are willing to accept it or at least review it, since I have personally moved on to Windows 11, where everything seems to be fixed. But I'm not fully sure it is, since I haven't had Windows 11 for too long, and I also have mostly switched over to use the Neovim built in terminal through the Neovide GUI, so I don't use Alacritty that much anymore.

**Comment by xy9485** [comment]
Any update on this issue? Mouse interaction(click and scroll) is still not working on Windows 21H1.

**Comment by kahnclusions** [comment]
@xy9485 I recommend trying [wezterm](https://github.com/wez/wezterm), it works great on Windows and has working mouse support.

**Comment by Viqsi** [comment]
> I have personally moved on to Windows 11, where everything seems to be fixed.

Can confirm that the workaround is no longer needed on Windows 11 21H2 and yet still necessary on Windows 10 21H2.

**Comment by Hasanabbas** [comment]
Would be nice to get this fixed for Windows 10. I'm on a corporate machine so God knows when we'll get Windows 11, and obviously I can't mess with System32

**Comment by Hasanabbas** [comment]
Yeah I read a bit more on it and realised this. Oh well

**Comment by jostaub** [comment]
may help some people:

i made a [branch](https://github.com/jostaub/alacritty/tree/wsl_experience_improvements) which integrates [#4501](https://github.com/alacritty/alacritty/pull/4501) from [fredizzimo](https://github.com/fredizzimo) and [commit 8d2d1eb](https://github.com/aytey/alacritty/commit/8d2d1eb780ed7b24004c9595595cc83aa131c795) from [aytey](https://github.com/aytey/alacritty/commits?author=aytey)
 which replaces "\r\n" with "\n" based on Version 0.10.1

**Comment by kirawi** [comment]
Wezterm has also compiled the fixed version of conpty: https://github.com/wez/wezterm/tree/main/assets/windows/conhost

**Comment by evertonse** [comment]
> I also have mostly switched over to use the Neovim built in terminal through the Neovide GUI, so I don't use Alacritty that much anymore.

It worked, thank you very much. I'm on windows 10 22h2.
I did almost exactly that. 
I've build your branch https://github.com/fredizzimo/alacritty/tree/add_openconsole_support 
Then i got the conpty.dll and OpenConsole.exe from WezTerm build, copied into the alacritty.exe folder 
and it simply worked, idk what you did, but thank you. 

As I'm not willing to got to W11 just now, and Alacritty is actually faster then Wezterm (on my machine it seems that way)
That saves me a lotta trouble. Thank you

**Comment by kchibisov** [comment]
Yeah, performance on windows is bottle necked on ConPty. Even macOS alacritty is 10-100x faster than windows build, simply because conpty is too slow.

**Comment by smprather** [comment]
I can't believe these guys are abandoning Win10 users. I'm stuck on a Win10 corp laptop, like many of us are. Just wanted to tag the thread that as of 10/1/23, the above work-around still works. I used @anupamsr 's [fork](https://github.com/anupamsr/alacritty/tree/add_openconsole_support). Looks like it had too many conflicts to merge into @fredizzimo 's fork (but I'm not GitWizard). Thanks!

**Comment by kchibisov** [comment]
@smprather it looks like you haven't looked into forks much or patches related. The required patches are merged and loading conpty.dll is supported on master.

**Comment by smprather** [comment]
@kchibisov Yeh, I'm a Git dunce. There can be no doubt. So to be clear: What's the best route for me to take to get latest Alacritty w/ mouse working on Win10?

**Comment by kchibisov** [comment]
build from master and add conpty.dll/OpenConsole.exe in the path.

**Comment by smprather** [comment]
Ok, built from master. I had conpty.dll/OpenConsole.exe (that I got from WezTerm) in the same dir as the fresh alacritty.exe (C:\Program Files\Alacritty). Tried it, but I was getting a freeze-up. So I name-changed them "out of the way" (added _ to names), re-ran Alacritty, and now everything works perfect. Including an ssh from WSL to another machine running tmux. Thanks @kchibisov / @fredizzimo / @anupamsr ! _(Note: I'm getting an occasional freeze-up. It's not a huge deal since I work via tmux, so I just close Alacritty and re-launch. I'm assuming it is unrelated and will be resolved eventually)_

**Comment by smprather** [comment]
Nope. The freeze-up only takes a minute or two of working to kick in. Hopefully it is resolved soon. Looks like I'm back to KiTTY again :(. Maybe I'll give WezTerm a shot.

**Comment by kchibisov** [comment]
just try older commits.

**Comment by smprather** [comment]
Energy is running too low to throw darts in the dark. I may try again next weekend. Even though @fredizzimo 's fork is way out of date, it actually works pretty good (mouse works, no freezes... so far). I'll probably be missing some new things like undercurl, OSC-52, etc. But I can tough it out till master stabilizes.

**Comment by anupamsr** [comment]
Use this one. I have been maintaining it for quite some time, though stopped around 4 months ago. I can update it if essential. https://github.com/anupamsr/alacritty/tree/add_openconsole_support

**Comment by smprather** [comment]
I think I tried that one last night and it didn't work. I'll try again and let you know. Definitely not "essential". This whole quest started when I saw LazyVim mention "undercurl". I think that's pretty good evidence of low-prio 😄.
I opened a separate issue on master branch freezing #7261 . Except for the freeze, it was working well.

**Comment by kchibisov** [comment]
I really don't understand this fuss with forks when the patch is upstream the way it was. it's all just dll loading in the end of the day, nothing special is required. Right here https://github.com/alacritty/alacritty/blob/ace987f343649ae98e5fb63cf825414855ccd86e/alacritty_terminal/src/tty/windows/conpty.rs#L49

It's likely just something got borked niside the upstream winit update. Just bisecting alacritty is more than enough to get this fixed, etc. Or maybe posting the `--print-events` output.

**Comment by kchibisov** [comment]
And be aware that all features are basically tied to those `dll` libs, if they update to have undercurl, alacritty will have all of that automatically, because `conpty` basically is a multiplexer and it filters out everything it doesn't understand.

**Comment by smprather** [comment]
I'm an electrical engineer, not a sw-dev. I can, and will, bisect to find the freeze-up commit. But I'm a git-dunce, so first, I'll have to figure out how to efficiently bisect. Then, each iteration is a 5 minute build followed by a few minutes of running to if freeze happens (I haven't identified the pattern yet). And these iterations must happen via awful Windows CLI environment. So that's another source of friction. I *will* get it done though.

**Comment by smprather** [comment]
I came for the undercurl, but I'm staying for the speed. Even working via a horrible 150ms ping to my ssh servers in Sweden, the GUI speed-up vs. KiTTY is non-negligible when doing nvim scrolling, tmux window cycling, etc.

**Comment by anupamsr** [comment]
> I really don't understand this fuss with forks when the patch is upstream the way it was. it's all just dll loading in the end of the day, nothing special is required. Right here
> 
> https://github.com/alacritty/alacritty/blob/ace987f343649ae98e5fb63cf825414855ccd86e/alacritty_terminal/src/tty/windows/conpty.rs#L49
> 
> It's likely just something got borked niside the upstream winit update. Just bisecting alacritty is more than enough to get this fixed, etc. Or maybe posting the `--print-events` output.

There is one old patch that adds support for openconsole (https://github.com/fredizzimo/alacritty) which was working well but it wasn't updated. I updated it and submitted the patch (https://github.com/fredizzimo/alacritty/pull/1). Neither my patch nor the old patch were merged. Later it borked because of winit which I also fixed if I remember correctly.

The reason I have not updated it in last 4 months is because there was a parallel commit done to the alacritty master achieving the same thing, which told me that noone has bothered to actually look at my repo that I had been maintaining.

For those who still want to work on Windows 10 like I do, I suggest to just clone my repo.

**Comment by tfriedel** [comment]
I just spent a rather long time figuring out this issue. I have it solved now thanks to the tip to copy conpty and openconsole to alacritty. I'd recommend to bundle them with alacritty on Windows, as no doubt many other people will either never figure this out or waste their time.

**Comment by abdenourgnx** [comment]
Hello @tfriedel, well i went all the way down to find your comment. can you briefly explain how you managed to make it work? thank you.

**Comment by tfriedel** [comment]
@abdenourgnx 
Sure! 
1. Go to https://github.com/wez/wezterm/tree/main/assets/windows/conhost and download [OpenConsole.exe](https://github.com/wez/wezterm/blob/main/assets/windows/conhost/OpenConsole.exe) and [conpty.dll](https://github.com/wez/wezterm/blob/main/assets/windows/conhost/conpty.dll)
2. Put these files into your alacritty program directory (e.g. c:\Program Files\Alacritty\ )

That's all! Alacritty will find these files and use them. (Unless you explicity disabled the conpty feature by setting winpty_backend to True)

**Comment by abdenourgnx** [comment]
@tfriedel Like magic. Thank you for saving me from hours of pain.

**Comment by 1812-Ram** [comment]
@tfriedel  Thanks mate!! work as treat!!

**Comment by tfriedel** [comment]
For anyone still having this issue, there's now a bundled version of alacritty with newer versions of conhost / openconsole and other fixes for windows around copy pasting issues:

https://github.com/tfriedel/alacritty/releases/tag/v0.16.1-windows-fixes

**Comment by FileFoundex** [comment]
> For anyone still having this issue, there's now a bundled version of alacritty with newer versions of conhost / openconsole and other fixes for windows around copy pasting issues:
> 
> https://github.com/tfriedel/alacritty/releases/tag/v0.16.1-windows-fixes

Thank you, I've spent hours trying to fix this
